{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.1",
    "cveMetadata": {
        "cveId": "CVE-2025-3121",
        "assignerOrgId": "1af790b2-7ee1-4545-860a-a788eba489b5",
        "state": "PUBLISHED",
        "assignerShortName": "VulDB",
        "dateReserved": "2025-04-02T13:35:42.010Z",
        "datePublished": "2025-04-02T22:00:12.390Z",
        "dateUpdated": "2025-04-03T19:26:25.202Z"
    },
    "containers": {
        "cna": {
            "providerMetadata": {
                "orgId": "1af790b2-7ee1-4545-860a-a788eba489b5",
                "shortName": "VulDB",
                "dateUpdated": "2025-04-02T22:00:12.390Z"
            },
            "title": "PyTorch torch.jit.jit_module_from_flatbuffer memory corruption",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "type": "CWE",
                            "cweId": "CWE-119",
                            "lang": "en",
                            "description": "Memory Corruption"
                        }
                    ]
                }
            ],
            "affected": [
                {
                    "vendor": "n/a",
                    "product": "PyTorch",
                    "versions": [
                        {
                            "version": "2.6.0",
                            "status": "affected"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "A vulnerability classified as problematic has been found in PyTorch 2.6.0. Affected is the function torch.jit.jit_module_from_flatbuffer. The manipulation leads to memory corruption. Local access is required to approach this attack. The exploit has been disclosed to the public and may be used."
                },
                {
                    "lang": "de",
                    "value": "Es wurde eine problematische Schwachstelle in PyTorch 2.6.0 entdeckt. Es geht dabei um die Funktion torch.jit.jit_module_from_flatbuffer. Mittels Manipulieren mit unbekannten Daten kann eine memory corruption-Schwachstelle ausgenutzt werden. Der Angriff hat dabei lokal zu erfolgen. Der Exploit steht zur öffentlichen Verfügung."
                }
            ],
            "metrics": [
                {
                    "cvssV4_0": {
                        "version": "4.0",
                        "baseScore": 4.8,
                        "vectorString": "CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N",
                        "baseSeverity": "MEDIUM"
                    }
                },
                {
                    "cvssV3_1": {
                        "version": "3.1",
                        "baseScore": 3.3,
                        "vectorString": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L",
                        "baseSeverity": "LOW"
                    }
                },
                {
                    "cvssV3_0": {
                        "version": "3.0",
                        "baseScore": 3.3,
                        "vectorString": "CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L",
                        "baseSeverity": "LOW"
                    }
                },
                {
                    "cvssV2_0": {
                        "version": "2.0",
                        "baseScore": 1.7,
                        "vectorString": "AV:L/AC:L/Au:S/C:N/I:N/A:P"
                    }
                }
            ],
            "timeline": [
                {
                    "time": "2025-04-02T00:00:00.000Z",
                    "lang": "en",
                    "value": "Advisory disclosed"
                },
                {
                    "time": "2025-04-02T02:00:00.000Z",
                    "lang": "en",
                    "value": "VulDB entry created"
                },
                {
                    "time": "2025-04-02T15:40:45.000Z",
                    "lang": "en",
                    "value": "VulDB entry last update"
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "value": "Default436352 (VulDB User)",
                    "type": "reporter"
                }
            ],
            "references": [
                {
                    "url": "https://vuldb.com/?id.303012",
                    "name": "VDB-303012 | PyTorch torch.jit.jit_module_from_flatbuffer memory corruption",
                    "tags": [
                        "vdb-entry",
                        "technical-description"
                    ]
                },
                {
                    "url": "https://vuldb.com/?ctiid.303012",
                    "name": "VDB-303012 | CTI Indicators (IOB, IOC, IOA)",
                    "tags": [
                        "signature",
                        "permissions-required"
                    ]
                },
                {
                    "url": "https://vuldb.com/?submit.525049",
                    "name": "Submit #525049 | pytorch pytorch (in torch.jit.jit_module_from_flatbuffer) torch 2.6.0 Memory Leak",
                    "tags": [
                        "third-party-advisory"
                    ]
                },
                {
                    "url": "https://github.com/pytorch/pytorch/issues/149800",
                    "tags": [
                        "issue-tracking"
                    ]
                },
                {
                    "url": "https://github.com/pytorch/pytorch/issues/149800#issue-2940240700",
                    "tags": [
                        "exploit",
                        "issue-tracking"
                    ]
                }
            ]
        },
        "adp": [
            {
                "metrics": [
                    {
                        "other": {
                            "type": "ssvc",
                            "content": {
                                "timestamp": "2025-04-03T19:20:19.683314Z",
                                "id": "CVE-2025-3121",
                                "options": [
                                    {
                                        "Exploitation": "poc"
                                    },
                                    {
                                        "Automatable": "no"
                                    },
                                    {
                                        "Technical Impact": "partial"
                                    }
                                ],
                                "role": "CISA Coordinator",
                                "version": "2.0.3"
                            }
                        }
                    }
                ],
                "title": "CISA ADP Vulnrichment",
                "providerMetadata": {
                    "orgId": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
                    "shortName": "CISA-ADP",
                    "dateUpdated": "2025-04-03T19:26:25.202Z"
                }
            }
        ]
    }
}